### PR TITLE
[SKIP-1164] Delete replace instana config with grafana-agent in defau…

### PR DESCRIPTION
**Pull request by martinhny**
…lt deny network policy

### CODE REVIEW

File: controllers/namespace/default_deny_network_policy.go [hunk 1]
```diff
@@ -7,21 +7,15 @@ import (
...
-	cmapNamespacedName := types.NamespacedName{Namespace: "skiperator-system", Name: "instana-networkpolicy-config"}
-	instanaConfigMap, err := util.GetConfigMap(r.GetClient(), ctx, cmapNamespacedName)
-
-	if !util.ErrIsMissingOrNil(r.GetRecorder(), err, "Cannot find configmap named instana-networkpolicy-config in namespace skiperator-system", namespace) {
-		return reconcile.Result{}, err
-	}
 
 	networkPolicy := networkingv1.NetworkPolicy{ObjectMeta: metav1.ObjectMeta{Namespace: namespace.Name, Name: "default-deny"}}
+	var err error
 	_, err = ctrlutil.CreateOrPatch(ctx, r.GetClient(), &networkPolicy, func() error {
```
Comment (anderssonw) [inline controllers/namespace/default_deny_network_policy.go:19]:
```suggestion
	_, err := ctrlutil.CreateOrPatch(ctx, r.GetClient(), &networkPolicy, func() error {
```

File: controllers/namespace/default_deny_network_policy.go [hunk 2]
```diff
@@ -88,13 +82,26 @@ func (r *NamespaceReconciler) reconcileDefaultDenyNetworkPolicy(ctx context.Cont
...
 							},
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{"app": "grafana-agent"},
+							},
+						},
+					},
+					Ports: []networkingv1.NetworkPolicyPort{
+						{
+							Protocol: util.PointTo(corev1.ProtocolTCP),
+							Port:     util.PointTo(intstr.FromInt(4317)),
```
Comment (anderssonw) [inline controllers/namespace/default_deny_network_policy.go:100]: where do the port numbers come from? haven't had a lot to do with grafana-agent

**Review by anderssonw — APPROVED** [review]

File: controllers/namespace/default_deny_network_policy.go (hunk 2 above)
Comment (martinhny) [reply controllers/namespace/default_deny_network_policy.go:100]: From here https://kartverket.atlassian.net/browse/SKIP-1164

File: controllers/namespace/default_deny_network_policy.go (hunk 1 above)
Comment (martinhny) [reply controllers/namespace/default_deny_network_policy.go]: fixed :)

File: controllers/namespace/default_deny_network_policy.go [hunk 3]
```diff
@@ -88,13 +81,26 @@ func (r *NamespaceReconciler) reconcileDefaultDenyNetworkPolicy(ctx context.Cont
...
 				{
 					To: []networkingv1.NetworkPolicyPeer{
 						{
-							IPBlock: &networkingv1.IPBlock{
-								CIDR: instanaConfigMap.Data["cidrBlock"],
+							NamespaceSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{"kubernetes.io/metadata.name": "grafana-agent"},
 							},
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{"app": "grafana-agent"},
```
[the remark this answers is not in the document]
Comment (martinhny) [reply controllers/namespace/default_deny_network_policy.go]: aha. changed

**Pull request merged**